src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java

```diff
@@ -167,13 +167,18 @@ public void modifyPlayer(String token, Long lobbyId, String newName, Boolean rea
         catch (DuplicateUserNameInLobbyException e) {
             throw new ResponseStatusException(HttpStatus.CONFLICT, "Username " + e.userName() + " is already taken.");
         }
+        catch (RegisteredUserLobbyNameChangeException e) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "User with id " + e.getIdRegisteredPlayer() + " is a registered player, " +
```

Comment:
I'm more a fan of setting the message where the actual exception is thrown.
/nitpicking

Reply:
Do you mean adding errorMessage to RegisteredUserLobbyNameChangeException and then setting new Response....Exception(...., e.getErrorMessage()); ?

Reply:
yep

Reply:
sure, I can do that

Reply:
I'm no expert in exceptions and it came to my mind just yesterday, and in our code base we do it 95% of the times like you did previously: but why would we want to mutate the error message in the service which basically should just translate it to a ResponseStatusException and not just set the correct error message right at the start?
That was my thought^^.
Overall: cool, tyvm!

Reply:
Those are different things.

- ResponseStatusException holds a message for the user.
- RegisteredUserLobbyNameChangeException is an internal representation of the error.

If I remember correctly, I was throwing ResponseStatusException directy (to avoid this translations) and you made me change it to the current pattern. Please make a consistent decision.

Reply:
I think we might want different error messages depending on the internal and external exceptions. As responseStatusException gets sent to the client, and we don't want to leak information.
But in this case it shouldn't matter as we set the errormessage to "User with id " *** + " is a registered player".

Reply:
> Those are different things.
> 
> * ResponseStatusException holds a message for the user.
> * RegisteredUserLobbyNameChangeException is an internal representation of the error.
> 

Yea, I think I will leave it like this then. To make it "future" proof. If someone changes the errormessage ot something we wouldn't want to send back

Reply:
> Those are different things.
> 
> * ResponseStatusException holds a message for the user.
> * RegisteredUserLobbyNameChangeException is an internal representation of the error.
> 
> If I remember correctly, I was throwing ResponseStatusException directy (to avoid this translations) and you made me change it to the current pattern. Please make a consistent decision.

You were throwing a ResponseStatusException directly in the Game class. This was wrong and is still wrong. The decision is consistent.
I was just talking about the Message.

Reply:
> I think we might want different error messages depending on the internal and external exceptions. As responseStatusException gets sent to the client, and we don't want to leak information. But in this case it shouldn't matter as we set the errormessage to "User with id " *** + " is a registered player".

Ah ok. Didn't think about that.